GliaWidgets/SwiftUI/Components/SnackBarView/SnackBar+Presenter.swift

```diff
@@ -0,0 +1,67 @@
+import Combine
+import SwiftUI
+import UIKit
+
+extension SnackBar {
+    final class Presenter {
+        weak var parentViewController: UIViewController?
+
+        let hostingController: UIHostingController<ContentView>
+        let updatePublisher: CurrentValueSubject<ContentView.ViewState, Never>
+
+        private var serialQueue = SerialQueue()
+
+        init(parentViewController: UIViewController, style: Theme.SnackBarStyle) {
+            self.parentViewController = parentViewController
+            let publisher = CurrentValueSubject<ContentView.ViewState, Never>(.disappear)
+            self.updatePublisher = publisher
+            self.hostingController = .init(
+                rootView: .init(
+                    style: style,
+                    publisher: publisher.eraseToAnyPublisher()
+                )
+            )
+        }
+
+        func add() {
+            guard let parent = parentViewController else { return }
+            remove()
+
+            parent.addChild(hostingController)
+            hostingController.willMove(toParent: parent)
+            parent.view.addSubview(hostingController.view)
+            hostingController.didMove(toParent: parent)
+            hostingController.view.backgroundColor = .clear
+
+            hostingController.view.translatesAutoresizingMaskIntoConstraints = false
+             NSLayoutConstraint.activate([
+                hostingController.view.topAnchor.constraint(greaterThanOrEqualTo: parent.view.topAnchor),
+                hostingController.view.bottomAnchor.constraint(equalTo: parent.view.bottomAnchor),
+                hostingController.view.leadingAnchor.constraint(equalTo: parent.view.leadingAnchor),
+                hostingController.view.trailingAnchor.constraint(equalTo: parent.view.trailingAnchor)
+             ])
+        }
+
+        func remove() {
+            parentViewController?.children
+                .filter { $0 as? UIHostingController<ContentView> != nil }
+                .forEach { $0.removeFromParent() }
+        }
+
+        func show(
+            text: String,
+            with offset: CGFloat
+        ) {
+            serialQueue.addOpeation(
+                .init { [weak self] done in
+                    self?.hostingController.rootView.offset = offset
+                    self?.updatePublisher.send(.appear(text))
+                    DispatchQueue.main.asyncAfter(deadline: .now() + .seconds(3)) { [weak self] in
```

Comment:
This delay is also should be achieved via reactive scheduler or injected closure to enable unit/snapshot testing. Let me know if you need help with it.

Reply:
fixed - https://github.com/salemove/ios-sdk-widgets/commit/f529ff1c062e6c8cf12a429d21cb10dc1f01d838